pkg/kudoctl/packages/types.go

```diff
@@ -0,0 +1,71 @@
+package packages
+
+import (
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+)
+
+const (
+	APIVersion = "kudo.dev/v1beta1"
+)
+
+// This is an abstraction which abstracts the underlying packages, which is likely file system or compressed file.
+// There should be a complete separation between retrieving a packages if not local and working with a packages.
+
+// Package is an abstraction of the collection of files that makes up a package.  It is anything we can retrieve the Resources from.
+type Package struct {
+	// transformed server view
+	Resources *Resources
+	// working with local package files
+	Files *Files
+}
+
+// Resources is collection of CRDs that are used when installing operator
+// during installation, package format is converted to this structure
+type Resources struct {
+	Operator        *v1beta1.Operator
+	OperatorVersion *v1beta1.OperatorVersion
+	Instance        *v1beta1.Instance
+}
+
+type Params []v1beta1.Parameter
+
+// Len returns the number of params.
+// This is needed to allow sorting of params.
+func (p Params) Len() int { return len(p) }
+
+// Swap swaps the position of two items in the params slice.
+// This is needed to allow sorting of params.
+func (p Params) Swap(i, j int) { p[i], p[j] = p[j], p[i] }
+
+// Less returns true if the name of a param a is less than the name of param b.
+// This is needed to allow sorting of params.
+func (p Params) Less(x, y int) bool {
+	return p[x].Name < p[y].Name
+}
+
+// Files represents the raw operator package format the way it is found in the tgz packages
+type Files struct {
+	Templates map[string]string
+	Operator  *Operator
+	Params    Params
```

Comment:
I think for consistency, Params here should be ParametersFile (and Operator should probably be renamed to OperatorFile).

Also we should probably unite on params vs parameters, but I am fine with all as a follow-up - I am just noting it here because I just stumbled upon that :)

Reply:
> Params here should be ParametersFile (and Operator to OperatorFile)

I'm not a big fan of this. Both already live in the `Files` struct so it feels redundant. If anything, I'd rather rename the `ParametersFile` to something better but I couldn't come up with a better name so far.

Reply:
Actually, now that I'm looking at it closely: why is `Files.Params` not the same as `ParameterFiles? 🤔

Reply:
It is the same thing minus the api version :) and since this struct contains operator full with api version, I think it should be the case for params as well... I mean, it's just a very minor nit :)

Reply:
`apiVersion` and the fact that params [are marked as required](https://github.com/kudobuilder/kudo/pull/1047/files#diff-1d0b14123ce824770ab6c4a21f1b3767R66) by default?

Reply:
It's `OperatorFile` and `ParamsFile`! Done.